crates/bin/prove_block/src/lib.rs

```diff
@@ -63,6 +64,9 @@ impl fmt::Display for ProveBlockError {
             ProveBlockError::TreeError(err) => write!(f, "Tree Error: {}", err),
             ProveBlockError::ContractClassError(err) => write!(f, "Contract Class Error: {}", err),
             ProveBlockError::SnOsError(err) => write!(f, "SnOs Error: {}", err),
+            ProveBlockError::LegacyContractDecompressionError(err) => {
+                write!(f, "Legacy class decompression Error: {}", err)
```

Comment:
I missed this enum in my review of @whichqua 's PR but we should definitely derive `thiserror::Error` for `ProveBlockError` instead of implementing this manually.